pkg/lib/bundle/validate.go

```diff
@@ -99,16 +98,57 @@ func (i imageValidator) ValidateBundleFormat(directory string) error {
 		validationErrors = append(validationErrors, err)
 	}
 
-	// Validate annotations.yaml
-	annotationsFile, err := ioutil.ReadFile(filepath.Join(annotationsDir, AnnotationsFile))
+	// Validate annotations file
+	files, err := ioutil.ReadDir(metadataDir)
 	if err != nil {
-		fmtErr := fmt.Errorf("Unable to read annotations.yaml file: %s", err.Error())
-		validationErrors = append(validationErrors, fmtErr)
+		validationErrors = append(validationErrors, err)
+	}
+
+	// Look for the metadata and manifests sub-directories to find the annotations file
+	fileAnnotations := &AnnotationMetadata{}
+	dependenciesFile := &registry.DependenciesFile{}
+	for _, f := range files {
```

Comment:
This loop seems to be dependent on the guarantee that `ioutil.ReadDir` sorts its return value by filename (so annotations.yaml always comes before dependencies.yaml), and that fileAnnotations won't be modified by DecodeFile when passed a dependencies file. Can it be refactored a bit to be clearer and to require fewer assumptions?